src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -459,27 +541,28 @@ private async Task ProcessIncomingAsync(ConnectionContext connection)
                 }
             }
             finally
-            {
-                keepAliveTimer.Stop();
-            }
+            {
+                keepAliveTimer.Stop();
+                _serviceConnectionRemoveTcs.TrySetResult(true);
+            }
         }
 
         private Task DispatchMessageAsync(ServiceMessage message)
-        {
-            switch (message)
-            {
-                case OpenConnectionMessage openConnectionMessage:
-                    return OnConnectedAsync(openConnectionMessage);
-                case CloseConnectionMessage closeConnectionMessage:
-                    return OnDisconnectedAsync(closeConnectionMessage);
-                case ConnectionDataMessage connectionDataMessage:
-                    return OnMessageAsync(connectionDataMessage);
-                case ServiceErrorMessage serviceErrorMessage:
-                    return OnServiceErrorAsync(serviceErrorMessage);
-                case PingMessage pingMessage:
-                    return OnPingMessageAsync(pingMessage);
-                case AckMessage ackMessage:
-                    return OnAckMessageAsync(ackMessage);
+        {

```

Comment:
Is this just a formatting change?

Reply:
I'm afraid so, I haven't touched these codes.

Reply:
I'm afraid so, I haven't touched these.